lib/plugins/aws/package/compile/events/websockets/index.js

```diff
@@ -51,9 +51,45 @@ class AwsCompileWebsockets {
       },
     };
 
-    // TODO: Complete schema, see https://github.com/serverless/serverless/issues/8019
     this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'websocket', {
-      anyOf: [{ type: 'string' }, { type: 'object' }],
+      anyOf: [
+        { type: 'string' },
+        {
+          type: 'object',
+          properties: {
+            route: { type: 'string' },
+            routeResponseSelectionExpression: {
+              type: 'string',
+              regexp: /^(?:\$default)$/.toString(),
+            },
+            authorizer: {
+              anyOf: [
+                { type: 'string' },
+                {
+                  type: 'object',
+                  properties: {
+                    name: { type: 'string' },
+                    identitySource: { type: 'array', items: { type: 'string' } },
+                  },
+                  required: ['name'],
+                  additionalProperties: false,
+                },
+                {
+                  type: 'object',
+                  properties: {
+                    arn: { type: 'string' },
+                    identitySource: { type: 'array', items: { type: 'string' } },
+                  },
+                  required: ['arn'],
+                  additionalProperties: false,
+                },
```

Comment:
Let's maybe use one object (it's not great that we duplicate `identitySource` definition), and express `name` and `arn` interchangeability with notation as:

https://github.com/serverless/serverless/blob/ce51c8fb6fea254affd51361b6a1cf551b5d8a36/lib/plugins/aws/package/compile/events/cloudWatchEvent/index.js#L38-L43

Reply:
done!